Formula/bcftools.rb

```diff
@@ -1,9 +1,11 @@
 class Bcftools < Formula
   desc "Tools for BCF/VCF files and variant calling from samtools"
   homepage "https://www.htslib.org/"
-  url "https://github.com/samtools/bcftools/releases/download/1.11/bcftools-1.11.tar.bz2"
-  sha256 "3ceee47456ec481f34fa6c34beb6fe892b5b365933191132721fdf126e45a064"
-  license "MIT"
+  url "https://github.com/samtools/bcftools/releases/download/1.12/bcftools-1.12.tar.bz2"
+  sha256 "7a0e6532b1495b9254e38c6698d955e5176c1ee08b760dfea2235ee161a024f5"
+  # The bcftools source code is MIT/Expat-licensed, but when it is configured
+  # with --enable-libgsl the resulting executable is GPL-licensed.
+  license "GPL-3.0"
```

Comment:
Does `bcftools` link statically with `gsl`? (If it does, it probably shouldn't.) I believe that's the only time linking affects the license. [1] Otherwise the listed license can stay MIT.

[1] There is some contention over this issue.

Reply:
It links dynamically with `gsl`. The bcftools [LICENSE](https://github.com/samtools/bcftools/blob/1.12/LICENSE) and [INSTALL](https://github.com/samtools/bcftools/blob/1.12/INSTALL#L132-L159) documentation make it clear that the bcftools authors consider that this makes the dynamically-linked `bcftools` executable GPL-licensed.

If it is the belief of the Homebrew maintainers that the listed license should stay MIT, then they should revert this part of the change.

Reply:
I'm happy to defer to the `bcftools` documentation on this. If it's their opinion their software is GPL-licensed when linking with `gsl` (even dynamically), then we should respect that.

However, I would like to resolve whether this is `-or-later` or `-only`.

Reply:
I may have time available to investigate this in due course. Alternatively you could resolve this question by checking the GSL source yourself; or the _gsl.rb_ license line was added by @whoiswillma in 8803565ddce998371c13f1f76bd2f9f2cf12bda6 so perhaps he is the local knowledgeable person.

Reply:
I added license information using an automated script last summer in order to bootstrap the license DSL. My apologies but I don't know much about `gsl` specifically